File: 240808/수열 뒤집기/invert-sequence.c
#include <stdio.h>

void reversed_list(int num_list[], int start, int end) {
    while (start < end) {
        int temp = num_list[start];
        num_list[start] = num_list[end];
        num_list[end] = temp;

        start++;
        end--;
    }
}

int main() {
    int N, K;
    scanf("%d %d", &N, &K);

    int A1, A2, B1, B2;
    scanf("%d %d", &A1, &A2);
    scanf("%d %d", &B1, &B2);

    int num_list[N];
    for (int i = 0; i < N; i++) {
        num_list[i] = i + 1;
    }

    for (int i = 0; i < K; i++) {
        reversed_list(num_list, A1 - 1, A2 - 1);
        reversed_list(num_list, B1 - 1, B2 - 1);
    }

    for (int i = 0; i < N; i++) {
        printf("%d\n", num_list[i]);
    }

    return 0;
}

File: 240809/수열 뒤집기/invert-sequence.py
N,K = map(int,input().split())
A1,A2 = map(int,input().split())
B1,B2 = map(int,input().split())

num_list = [i + 1 for i in range(N)]

def reversed_list(num_list,start,end):
    while start < end:
        num_list[start],num_list[end] = num_list[end],num_list[start]

        start += 1
        end -= 1

for _ in range(K%12):
    reversed_list(num_list,A1-1,A2-1)
    reversed_list(num_list,B1-1,B2-1)

print('\n'.join(list(map(str,num_list))))